### PR TITLE
fix wal config not work on config.yml

### PR DESCRIPTION
**Pull request by Wing924**
Signed-off-by: Wing924 <weihe924stephen@gmail.com>

**What this PR does**:
Fix bug: `walconfig` on config file don't work

YAML library use reflect to assign fields, but reflect can't access private fields.
This PR make all ingester.WALConfig public so that `walconfig.*` can work without error.

**Which issue(s) this PR fixes**:

**Checklist**
- [X] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/ingester/wal.go [hunk 1]
```diff
@@ -29,21 +29,21 @@ import (
 
 // WALConfig is config for the Write Ahead Log.
 type WALConfig struct {
-	walEnabled         bool          `yaml:"wal_enabled,omitempty"`
-	checkpointEnabled  bool          `yaml:"checkpoint_enabled,omitempty"`
-	recover            bool          `yaml:"recover_from_wal,omitempty"`
-	dir                string        `yaml:"wal_dir,omitempty"`
-	checkpointDuration time.Duration `yaml:"checkpoint_duration,omitempty"`
+	WalEnabled         bool          `yaml:"wal_enabled,omitempty"`
```
Comment (codesome) [inline pkg/ingester/wal.go]: `WALEnabled`

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,7 @@
 * [ENHANCEMENT] Experimental TSDB: Added dedicated flag `-experimental.tsdb.bucket-store.tenant-sync-concurrency` to configure the maximum number of concurrent tenants for which blocks are synched. #2026
 * [ENHANCEMENT] Experimental TSDB: Expose metrics for objstore operations (prefixed with `cortex_<component>_thanos_objstore_`, component being one of `ingester`, `querier` and `compactor`). #2027
 * [BUGFIX] Experimental TSDB: fixed `/all_user_stats` and `/api/prom/user_stats` endpoints when using the experimental TSDB blocks storage. #2042
+* [BUGFIX] Ingester: fixed `ingester.walconfig` not work on config.yml. #2071
```
Comment (pracucci) [inline CHANGELOG.md]: I would rephrase it to make it a bit more clear, something like:

```
* [BUGFIX] Fixed parsing of the WAL configuration when specified in the YAML config file. #2071
```

File: pkg/ingester/ingester.go [hunk 3]
```diff
@@ -141,7 +141,7 @@ func New(cfg Config, clientConfig client.Config, limits *validation.Overrides, c
 		return NewV2(cfg, clientConfig, limits, registerer)
 	}
 
-	if cfg.WALConfig.walEnabled {
+	if cfg.WALConfig.WalEnabled {
```
Comment (pracucci) [inline pkg/ingester/ingester.go]: May you rename `WalEnabled` to `WALEnabled `, please?

File: pkg/ingester/wal.go [hunk 4]
```diff
@@ -29,21 +29,21 @@ import (
...
+	CheckpointEnabled  bool          `yaml:"checkpoint_enabled,omitempty"`
+	Recover            bool          `yaml:"recover_from_wal,omitempty"`
+	Dir                string        `yaml:"wal_dir,omitempty"`
+	CheckpointDuration time.Duration `yaml:"checkpoint_duration,omitempty"`
 	metricsRegisterer  prometheus.Registerer
```
Comment (pracucci) [inline pkg/ingester/wal.go]: Despite correct, please also add `yaml:"-"` to enforce we don't want it to be configurable.
